pkg/controllers/observability/tempo_components.go

```diff
@@ -56,27 +66,90 @@ func tempoSecretName(name string) string {
+			tempoSecret.Data["region"] = []byte(instance.Spec.Storage.ObjectStorageSpec.S3.Region)
+		}
+	} else if instance.Spec.Storage.ObjectStorageSpec.S3STS != nil {
+		tempoSecret.Data = map[string][]byte{
+			"bucket":   []byte(instance.Spec.Storage.ObjectStorageSpec.S3STS.Bucket),
+			"role_arn": []byte(instance.Spec.Storage.ObjectStorageSpec.S3STS.RoleARN),
+			"region":   []byte(instance.Spec.Storage.ObjectStorageSpec.S3STS.RoleARN),
+		}
+	} else if instance.Spec.Storage.ObjectStorageSpec.S3CCO != nil {
+		tempoSecret.Data = map[string][]byte{
+			"bucket": []byte(instance.Spec.Storage.ObjectStorageSpec.S3CCO.Bucket),
+			"region": []byte(instance.Spec.Storage.ObjectStorageSpec.S3CCO.Region),
+		}
+	} else if instance.Spec.Storage.ObjectStorageSpec.Azure != nil {
+		accountKeySecret := &corev1.Secret{}
+		err := k8sClient.Get(ctx, client.ObjectKey{
+			Namespace: operatorNamespace,
+			Name:      instance.Spec.Storage.ObjectStorageSpec.Azure.AccountKey.SecretName,
+		}, accountKeySecret)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get Azure account key secret %s: %w", instance.Spec.Storage.ObjectStorageSpec.Azure.AccountKey.SecretName, err)
+		}
+
 		tempoSecret.Data = map[string][]byte{
-			"bucket":            storageSecret.Data["bucket"],
-			"endpoint":          storageSecret.Data["endpoint"],
-			"access_key_id":     storageSecret.Data["access_key_id"],
-			"access_key_secret": storageSecret.Data["access_key_secret"],
+			"container":    []byte(instance.Spec.Storage.ObjectStorageSpec.Azure.Container),
+			"account_name": []byte(instance.Spec.Storage.ObjectStorageSpec.Azure.AccountName),
+			"account_key":  accountKeySecret.Data[instance.Spec.Storage.ObjectStorageSpec.Azure.AccountKey.Key],
+		}
+	} else if instance.Spec.Storage.ObjectStorageSpec.AzureWIF != nil {
+		tempoSecret.Data = map[string][]byte{
+			"container":    []byte(instance.Spec.Storage.ObjectStorageSpec.AzureWIF.Container),
+			"account_name": []byte(instance.Spec.Storage.ObjectStorageSpec.AzureWIF.AccountName),
+			"client_id":    []byte(instance.Spec.Storage.ObjectStorageSpec.AzureWIF.ClientID),
+			"tenant_id":    []byte(instance.Spec.Storage.ObjectStorageSpec.AzureWIF.TenantID),
+		}
+		if instance.Spec.Storage.ObjectStorageSpec.AzureWIF.Audience != "" {
+			tempoSecret.Data["audience"] = []byte(instance.Spec.Storage.ObjectStorageSpec.AzureWIF.Audience)
+		}
+	} else if instance.Spec.Storage.ObjectStorageSpec.GCS != nil {
+		accountKeySecret := &corev1.Secret{}
+		err := k8sClient.Get(ctx, client.ObjectKey{
+			Namespace: operatorNamespace,
+			Name:      instance.Spec.Storage.ObjectStorageSpec.GCS.KeyJSON.SecretName,
+		}, accountKeySecret)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get GCS keyJSON secret %s: %w", instance.Spec.Storage.ObjectStorageSpec.GCS.KeyJSON.SecretName, err)
+		}
+
+		tempoSecret.Data = map[string][]byte{
+			"bucket":   []byte(instance.Spec.Storage.ObjectStorageSpec.GCS.Bucket),
```

Comment:
The secret created for Tempo in the cluster-observability namespace need to have bucketname as the key instead of bucket.